java-checks/src/main/java/org/sonar/java/checks/CatchUsesExceptionWithContextCheck.java

```diff
@@ -78,11 +83,42 @@ public void scanFile(JavaFileScannerContext context) {
     if (context.getSemanticModel() != null) {
       scan(context.getTree());
     }
+    excludedCatchTrees.clear();
+  }
+
+  @Override
+  public void visitTryStatement(TryStatementTree tree) {
+    if (containsEnumValueOf(tree.block())) {
+      tree.catches().stream()
+        .filter(c -> c.parameter().symbol().type().is("java.lang.IllegalArgumentException"))
+        .findAny()
+        .ifPresent(excludedCatchTrees::add);
+    }
+    super.visitTryStatement(tree);
+  }
+
+  private static boolean containsEnumValueOf(Tree tree) {
+    EnumValueOfVisitor visitor = new EnumValueOfVisitor();
```

Comment:
Visitor should probably cut on anonymous classes and lambda bodies

Reply:
Correct, I have been lazy on this one.